importance-of-architecture.md

```diff
@@ -0,0 +1,36 @@
+---
+title: Avoid the urgency trap 
+subtitle: Or: Why architecture matters in software projects
+domain: campzulu.hashnode.dev
+tags: architecture, software-architecture, design-and-architecture
+cover: https://cdn.hashnode.com/res/hashnode/image/upload/v1665328931237/kXGgAFOdK.jpg?w=1600&h=840&fit=crop&crop=entropy&auto=compress,format&format=webp
+publishAs: romanutti
+ignorePost: true
+---
+
+A software project generally has different stakeholders.
+
+One obvious stakeholder is the customer. They commission software developers to build software that meets certain requirements. The customer's focus often lies on the **behavior** of the software: It should behave in a way that makes or saves money for the customer. Many software engineers consider this their main task. They believe their job is to implement functionality to meet businesses' requirements.
+
+However, this ignores that developers are their software's stakeholders too. Software contains the term *soft*. It is intended to be soft: Changing software and their behavior should be easy - otherwise, we would have called it *hard*ware. When the business changes a requirement or aims for a new feature, that change should be easily accomplishable. We use **architecture** to structure our applications to minimize the efforts required to build and maintain a system. Because architectural aspects often require a technical understanding of the system, it is the task of the developer (and architect) to ensure architecture in a software project.
+
+Both, implementing the behavior of the software and enforcing its architectural aspects require resources. As time and money are limited in software projects (at least those of which I have been a part), we ask ourselves which one to prioritize.
+
+# The Mere-Urgency Effect a.k.a. Why We're Bad at Prioritization
```

Comment:
Small SEO hint: h1 should reflect the title.

https://developers.google.com/style/headings#formatting-a-heading-or-title